mssqlcli/main.py

```diff
@@ -594,14 +595,40 @@ def refresh_completions(self, history=None, persist_priorities='all'):
         :param persist_priorities: 'all' or 'keywords'
         """
 
+        mssqlclclient_completion_refresher = self.get_completion_refresher_mssqlcliclient(
+                                                self.mssqlcliclient_query_execution)
+
         callback = functools.partial(self._on_completions_refreshed,
                                      persist_priorities=persist_priorities)
 
-        self.completion_refresher.refresh(self.mssqlcliclient_query_execution,
-                                          callback, history=history, settings=self.settings)
+        self.completion_refresher.refresh(mssqcliclient=mssqlclclient_completion_refresher,
+                                          callbacks=callback,
+                                          history=history,
+                                          settings=self.settings)
+
+
+
         return [(None, None, None,
                  'Auto-completion refresh started in the background.')]
 
+    def get_completion_refresher_mssqlcliclient(self, mssqlcliclient):
+        mssqlcliclient_completion_refreshser = MssqlCliClient(

```

Comment:
nit: naming - since we're creating a new MssqlCliClient, can we call this method create_completion_refresher_mssqlcliclient

Reply:
Fixed.